paasta_tools/cli/cmds/local_run.py

```diff
@@ -750,14 +679,20 @@ def run_docker_container(
         chosen_port = pick_random_port(service)
     environment = instance_config.get_env_dictionary()
     if not skip_secrets:
-        secret_environment = decrypt_secret_environment_variables(
-            secret_provider_name=secret_provider_name,
-            environment=environment,
-            soa_dir=soa_dir,
-            service_name=service,
-            cluster_name=instance_config.cluster,
-            secret_provider_kwargs=secret_provider_kwargs,
-        )
+        try:
+            secret_environment = decrypt_secret_environment_variables(
+                secret_provider_name=secret_provider_name,
+                environment=environment,
+                soa_dir=soa_dir,
+                service_name=service,
+                cluster_name=instance_config.cluster,
+                secret_provider_kwargs=secret_provider_kwargs,
+            )
+        except Exception:
```

Comment:
catch secret-specific error instead of `Exception` and print error details here instead of in util module